app/controllers/candidate_interface/references/selection_controller.rb

```diff
@@ -6,24 +6,80 @@ def new
           application_form: current_application,
           selected: current_application.application_references.selected.pluck(:id),
         )
+        @enough_references_provided = current_application.minimum_references_available_for_selection?
       end
 
       def create
         @selection_form = CandidateInterface::Reference::SelectionForm.new(selection_params)
+        @enough_references_provided = current_application.minimum_references_available_for_selection?
+
         if @selection_form.save!
           track_validation_error(@section_complete_form)
-          redirect_to candidate_interface_references_review_path
+          redirect_to candidate_interface_review_selected_references_path
+        elsif !@enough_references_provided
+          flash.now[:warning] = I18n.t('application_form.references.review.need_two')
+          render :new and return
         else
+          track_validation_error(@selection_form)
           render :new
         end
       end
 
+      def review
+        @references_selected = current_application.application_references.includes(:application_form).selected
+        @section_complete_form = SectionCompleteForm.new(completed: current_application.references_completed)
+
+        @selected_references_rows = [
+          {
+            key: 'Selected references',
+            value: reference_values,
+            action: 'Change selected references',
+            change_path: candidate_interface_select_references_path,
+          },
+        ]
+      end
+
+      def complete
+        @section_complete_form = SectionCompleteForm.new(completed: section_complete_params[:completed])
+
+        if !@section_complete_form.valid?
+          track_validation_error(@section_complete_form)
+          render :review
```

Comment:
`and return`

will also need the same instance vars assigned as in the `review` action in order to render without error